src/sage/combinat/symmetric_group_algebra.py

```diff
@@ -2068,8 +2068,152 @@ def dft(self, form=None, mult='l2r'):
+            assert F.is_finite()
+            assert F.order().is_square()
+            if F.characteristic().divides(G.cardinality()):
+                raise NotImplementedError("Not implemented when p|n!. Dimension of invariant forms may be greater than one and sqrt{d_rho/|G|} is not defined. See modular DFT.")
+            q = sqrt(F.order())
+
+            def invariant_symmetric_bilinear_matrix(partition):
+                specht_module = self.specht_module(partition)
+                rho = specht_module.representation_matrix
+                d_rho = specht_module.dimension()
+                R = PolynomialRing(F,'u',d_rho**2)
+                U_vars = R.gens()
+                U = matrix(R,d_rho,d_rho,U_vars)
+
+                def augmented_matrix(g):
+                    rho_g = rho(g)
+                    equation_matrix = rho_g.transpose()*U*rho_g.conjugate() - U
+                    augmented_system = []
+                    for i in range(d_rho):
+                        for j in range(d_rho):
+                            linear_expression = equation_matrix[i, j]
+                            row = [linear_expression.coefficient(u) for u in U_vars]
+                            augmented_system.append(row)
+                    return matrix(F, augmented_system)
+                total_system = matrix(F,0,d_rho**2)
+                for g in G:
+                    total_system = total_system.stack(augmented_matrix(g))
+                null_space = total_system.right_kernel()
+                U_mats = [matrix(F,d_rho,d_rho,b) for b in null_space.basis()]
+                return U_mats
+
+            def conj_square_root(u):
+                if u == 0:
+                    return 0
+                z = F.multiplicative_generator()
+                k = u.log(z)
+                if k % (q+1) != 0:
+                    raise ValueError("Unable to factor: u is not in base field GF(q)")
+                return z**((k//(q+1)) % (q-1))
+
+            def base_change_hermitian(U):
+                Up = U.LU()[2]
+                D = Up.diagonal()
+                A = ~Up * matrix.diagonal([d.sqrt() for d in D])
+                diag = (A.H * U * A).diagonal()
+                factor_diag = diagonal_matrix([conj_square_root(d) for d in diag])
+                return factor_diag*A.inverse()
+
+            def hat(f,partition):
+                specht_module = self.specht_module(partition)
+                rho = specht_module.representation_matrix
+                U = invariant_symmetric_bilinear_matrix(partition)[0]
+                A = base_change_hermitian(U)
+                sqrt_unitary_factor = sqrt(F(specht_module.dimension()/G.cardinality()))
+                return sqrt_unitary_factor*sum(f(g)*A*rho(g)*A.inverse() for g in G)
+
+            delta = lambda s: lambda t: 1 if t == s else 0
+            fourier_transform = [flatten([hat(delta(g),partition).list() for partition in Partitions(G.degree())]) for g in G]
```

Comment:
```suggestion
            def hat(g, partition):
                specht_module = self.specht_module(partition)
                rho = specht_module.representation_matrix
                U = invariant_symmetric_bilinear_matrix(partition)[0]
                A = base_change_hermitian(U)
                sqrt_unitary_factor = sqrt(F(specht_module.dimension() / G.cardinality()))
                return sqrt_unitary_factor * f(g) * A * rho(g) * A.inverse()

            PGdeg = Partitions(G.degree())
            fourier_transform = [flatten([hat(g, partition).list() for partition in PGdeg]) for g in G]
```
Same comment as the char 0 case. Although it would likely be better to cache those square root computations somehow.